### DISCUSSION TITLE
Well, the documentation is not enough I guess?

### DISCUSSION THREAD

**Discussion by Olyno** [supabase/supabase#438] [Questions] [answered]
Hi 👋🏻 

**Spoiler alert:** the documentation is great. Supabase's documentation is really good, and we learn a lot of things using it. **BUT**, a lot of stuff is missing in, creating confusions like in #437.

So, I decided to create a thread about a lot of questions I got myself when using Supabase (and ``supabase-js``):

1) What is a SupabaseClient in fact? Is it just a way to link Supabase and the project I did? Or is it a real client (1 client/user), and I need to create another one each time there is a new user?
2) The ``SupabaseClient.auth#signOut`` method from the ``supabase-js`` lib is really confusing me: "Signs out the current user, if there is a logged in user". But if we're using Supabase in a NodeJs backend like I do, what is exactly the current user? Because there is more than 1 user in this case.
3) Same for the ``SupabaseClient.auth#user`` method: "Get the JSON object for the logged in user."
4) How do I logout a specific user? I guess it's not implemented yet, so this one becomes a suggestion.
5) How do JWT tokens work? Like after each request I get one, nice, but how am I supposed to do? Is there any way to link my backend request object to the ``supabase-js`` lib?
6) "For security purposes, the auth schema is not exposed on the auto-generated API", well it's kinda annoying if I need to get some data using my backend, how am I supposed to do? :/

Anyway, thanks a lot for your work, Supabase is really great and I'm starting to use it a lot to give the most feedback I can :p

**Comment by GodGardensGuns** [comment, marked as answer]
Discussion #437 was started by my teammate. We also love what Supabase offers and are hoping to learn more about how it works in order to use it in production without fear of the unknown.

In response to some of your points... 
1. I have wondered this as well. I'm curious if having many users with active clients will use up limited connection slots on the database and require something like connection pooling.

2 - 5. JWT's are JSON objects that hold unencrypted user data alongside a cryptographic signature. The token is used to verify requests to the DB. Looks like supabase auto stores the token in HttpCookies on the client side (standard practice), but I'm not quite sure as I haven't looked into it. The client's token becomes invalidated after a set period of time, at which point they'd have to log in again.

I'd be interested in knowing how to refresh tokens, how long they have before they become "stale" (invalidated), and more about the JWT auth process on general. 

6. I believe the JWT is used to authenticate requests to the database. The intro guide recommends creating a public users table that auto-generates an entry for every new user in the auth.user table. Here's some more info with a useful query to get that done: https://supabase.io/docs/guides/auth#create-a-publicusers-table

**Reply by Olyno** [reply]
Thank you for taking the time to answer me!
Regarding my question about the "auth.users" table, I found it especially illogical to have to create a new table to publicly store the information provided to us in the "auth.users" table. Maybe it would be interesting to give access to it if the client created has the key "service_role"?

**Reply by GodGardensGuns** [reply]
The service role key seems to be for testing purposes only, as it can basically pass all row-level restrictions in the database.

Perhaps the way Supabase has it set up (where you can't really create/read data in the auth.users table) has something to do with encryption going on behind the scenes. Like I said, I would love to learn more from someone on the team/in the know.

**Reply by soedirgo** [reply]
Thanks for answering @dayvista!

On 1, since each user (i.e. your user) has different credentials, you need a client per user. Supabase uses PostgREST under the hood, which has [connection pooling](https://postgrest.org/en/v7.0.0/configuration.html#db-pool), so there's no need to worry about creating many clients.

Re: the Auth stuff, I'll let other members of the team who knows more to answer :)

**Reply by Olyno** [reply]
Well I guess you answered to all my questions by answering only at the first ahaha, thanks a lot 😄

**Reply by kiwicopple** [reply]
hey @Olyno . this is amazing feedback - thanks! I've added an issue [here](https://github.com/supabase/supabase/issues/492) and we'll improve the docs for all of your points

**Comment by albas99** [comment]
I'm currently facing the same issue here, being new to supabase the documentation as it is right now is making me feel like I have to be an expert developer to be able to piece together and use the information given in the documentation. It's definitely not beginner friendly for the most part. For example I'm using the python client and trying out some basic stuff like adding users programmatically but in the documentation there's nowhere error handling is mentioned to be able to retrieve feedback and know whether and operation on the database is successful or not, usually with something like API requests I can get the status code but so far in the documentation I can't find anything like that, only a page with the list of HTTP status codes.
